Source/player.cpp

```diff
@@ -515,26 +515,44 @@ void StartSpell(int pnum, Direction d, int cx, int cy)
 		return;
 	}
 
+	// Checks conditions for spell again, cause initial check was done when spell was queued and the parameters could be changed meanwhile
+	bool isValid = true;
+	switch (player.queuedSpell.spellType) {
+	case RSPLTYPE_SKILL:
+	case RSPLTYPE_SPELL:
+		isValid = CheckSpell(pnum, player.queuedSpell.spellId, player.queuedSpell.spellType, true) == SpellCheckResult::Success;

```

Comment:
With the previous issue out of the way, I have one last issue to report here. Mana isn't synced between clients, so if a player joins a game with 0 mana, their spells will always fail on remote clients for the duration of the game. However, this one should be easy to solve by syncing mana in `TPktHdr` structs the way we do for HP.

Reply:
I opened #4207 for this. 🙂